contributors/design-proposals/node/node-fence.md

```diff
@@ -0,0 +1,398 @@
+  metadata:
+   name: fence-method-template-eaton-pdu
+   namespace: default
+  data:
+   template.properties: |-
+          name=eaton_pdu
+          agent_name=fence_eaton_snmp
+          must_sucess=yes
+          ipaddr=192.168.1.4
+          password=eaton_password
+          username=eaton_admin
+          snmp-priv-prot=AES
+          snmp-priv-passwd=eaton_snmp_passwd
+          snmp-sec-level=authPriv
+          inet4-only=true
+```
+
+Method for enable and disable of brocade for host0:
+
+```
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-method-eaton-off-host0
+   namespace: default
+  data:
+   method.properties: |-
+        template=fence-method-template-eaton-pdu
+        plug=1
+        action=off
+
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-method-eaton-on-host0
+   namespace: default
+  data:
+   method.properties: |-
+        template=fence-method-template-eaton-pdu
+        plug=1
+        action=on
+```
+
+Method for enable and disable of eaton for host1:
+
+```
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-method-eaton-off-host1
+   namespace: default
+  data:
+   method.properties: |-
+        template=fence-method-template-eaton-pdu
+        plug=2
+        action=off
+
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-method-eaton-on-host1
+   namespace: default
+  data:
+   method.properties: |-
+        template=fence-method-template-eaton-pdu
+        plug=2
+        action=on
+```
+
+Same can be defined APC and eaton for both nodes.
+
+Finally fence configs:
+
+```
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-config-lago-kube-host0
+   namespace: default
+  data:
+   config.properties: |-
+    node_name=lago-kube-host0
+    isolation=fc-off
+    power_management=eaton-off eaton-on
+    recovery=fc-on
+
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+   name: fence-config-host1
+  data:
+   config.properties: |-
+    node_name=host
+    isolation=fc-off
+    power_management=eaton-off eaton-on
+    recovery=fc-on
+```
+
+
+## Aditional implementations details
+### Cluster fence policies
+Fencing policy allows each cluster to behave differently in case of connectivity issues. One of the main motivation for that is to prevent “fencing storms” from happening, but there are others. The controller is responsible to force the fence policy
+
+Fencing storm is a situation in which fencing for a few nodes in the cluster is triggered at the same time, due to an environmental issue.
+
+Examples:
+1. Switch failure - a switch used to connect a few nodes to the environment is failing, and there is no redundancy in the network environment. In such a case, the nodes will be reported as unresponsive, while they are still alive and kicking, and perhaps providing service through other networks they are attached to. If we fence those nodes, a few services will be offline until restarted, while it might not have been necessary.
```

Comment:
Sort of, there are downsides either way (deciding whether to fence or not).  Have a read of http://blog.clusterlabs.org/blog/2018/two-node-problems

The problem is that you have no way to know if it is a switch failure.
So either:
- you assume it's not a network failure, initiate fencing and potentially kill a healthy Pod (does it really count as healthy if no-one can reach it though) so it can be recovered, or
- you assume it is a network failure, don't initiate fencing and risk that there are multiple copies of some Pods